honeybee/aperturetype.py

```diff
@@ -0,0 +1,58 @@
+# coding=utf-8
+"""Aperture Types."""
+
+
+class _ApertureType(object):
+    __slots__ = ()
+
+    @property
+    def name(self):
+        return self.__class__.__name__
+
+    def to_dict(self):
+        """ApertureType as a dictionary."""
+        ap_type_dict = {
+            'type': self.name}
+        return ap_type_dict
+
+    def duplicate(self):
+        """Get a copy of this aperture type."""
+        return self.__copy__()
+
+    def ToString(self):
+        return self.__repr__()
+
+    def __copy__(self):
+        return self
+
+    def __eq__(self, other):
+        return self.__class__ == other.__class__
+
+    def __ne__(self, other):
+        return not self.__eq__(other)
+
+    def __repr__(self):
+        return self.name
+
+
+class Window(_ApertureType):
+    """Type for static apertures that are not intended to be opened."""
+    __slots__ = ()
+    pass
+
+
+class _Types(object):
+    """Aperture types."""
+
+    def __init__(self):
+        self._window = Window()
+
+    @property
+    def window(self):
+        return self._window
+
+    def __contains__(self, value):
+        return isinstance(value, _ApertureType)
+
+
+aperture_types = _Types()
```

Comment:
I think we should go for the Singleton implementation for types so we can remove cases like this. I don't think this change is part of this PR but something that we should consider to do in the future.

Reply:
I think this merits some consideration but it might be tough if we ever want to include properties on the types.  For example, it might be nice to store the `fraction_operable` on an `OperableWindow` type in honeybee-energy.  This way, the operable fraction property only gets created for operable windows and doesn't need to exist on the energy properties of every single aperture.  Let's consider this in more detail later.

Reply:
I agree. Let's keep this for later.